src/middlewares/parse-request-body.ts

```diff
@@ -0,0 +1,28 @@
+import { ExtendableContext, Next, Request } from 'koa'
```

Comment:
This middleware does the following:
- Take zod schema as param
- Parse incoming request body using the schema
- 400 if failure (with some error msg describing the errors)
- sets the parsed request body on the the context.request.body, so that the consumer now have access to the typed body.